lib/matplotlib/cbook/__init__.py

```diff
@@ -367,6 +367,18 @@ def process(self, s, *args, **kwargs):
                     self._remove_proxy(proxy)
 
 
+class EventEmitter(object):
```

Comment:
I am 50/50 on adding this.  It is nice to have, but if we add it, we should use it across all of the things that emit.